modules/hp_articles/hp_articles.module

```diff
@@ -63,6 +62,39 @@ function hp_articles_preprocess_paragraph__hp_articles(&$variables) {
     $filter = $variables['elements']['#paragraph']->field_hp_af_article_category_fil->target_id;
   }
 
+  $article_count = NULL;
+  if ($variables['elements']['#paragraph']->field_hp_af_article_count->value) {
+    $article_count = $variables['elements']['#paragraph']->field_hp_af_article_count->value;
+  }
+
+  $config = \Drupal::service('config.factory')->getEditable('hp_articles.settings');
+  $config->set('hp_article_count', $article_count)
+    ->save();
+
   $variables['hp_articles_feed'] = views_embed_view('hp_articles_ks_feed', 'hp_articles_block', $filter);
 
 }
+
+/**
+ * Implements hook views_pre_view().
+ * Display set number of articles.
+ */
+function hp_articles_views_pre_view(ViewExecutable $view) {
+  $config = \Drupal::config('hp_articles.settings');
+  $article_count = $config->get('hp_article_count');
+
+  if ($view->id() == 'hp_articles_ks_feed' && $view->current_display == 'hp_articles_block') {
+    $view->setItemsPerPage($article_count);
+    $pager = $view->display_handler->getOption('pager');
+    $pager['type'] = 'some';
+    $view->display_handler->setOption('pager', $pager);
+  }
+}
+
```

Comment:
May be a lot simpler to do something like response 1 here: https://drupal.stackexchange.com/questions/130082/views-pass-argument-for-number-of-records-to-display-to-pager-contextual-filter

Then you could simply pass another argument in the views_embed_view('hp_articles_ks_feed', 'hp_articles_block', [$filter,$count]) (untested, just an example), then get the $count you passed in the pre_build. Would allow you to set the count in the paragraph, not a global var, and this make it be able to use different counts for multiple widgets.